pkg/api/persistence/persistence.go

```diff
@@ -206,6 +248,29 @@ func (db Persistence) GetCheckByID(id uuid.UUID) (api.Check, error) {
 	return c, err
 }
 
+// GetScanStatus returns a scan struct with only the fields: Status, CheckCount, ChecksFinished
+// filled.
+func (db Persistence) GetScanStatus(ID uuid.UUID) (api.Scan, error) {
+	query := `SELECT data->>'status' as status, 
+	data->'checks_finished' as checks_finished,
+	data->'check_count' as check_count FROM scans WHERE id=?`
+	s := new(api.Scan)
+	var (
+		count, finished int
+		status          string
+	)
+	rest := []interface{}{&status, &count, &finished}
```

Comment:
Is this order correct according to the query?

Reply:
It is not thanks for pointing it out!! Also even If I added a test for this method, the values were also swapped in the "want" values of the test 😮  so it was passing!!